lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb

```diff
@@ -15,16 +16,56 @@ class FirebaseAppDistributionAction < Action
       FIREBASECMD_ACTION = "appdistribution:distribute".freeze
       BASE_URL = "https://firebaseappdistribution.googleapis.com"
       PATH = "/v1alpha/apps/"
+      MAX_POLLING_RETRIES = 60
+      POLLING_INTERVAL_S = 2
 
       extend Helper::FirebaseAppDistributionHelper
 
       def self.run(params)
         params.values # to validate all inputs before looking for the ipa/apk
-        validate_app!(params[:app])
+        platform = Actions.lane_context[Actions::SharedValues::PLATFORM_NAME]
+        binary_path = params[:ipa_path] || params[:apk_path]
+
+        if params[:app] # Set app_id if it is specified as a parameter
+          app_id = params[:app]
+        elsif platform == :ios
```

Comment:
Should there be an elseif for android? Or is that implicit with `params[:app]`

Reply:
So this code is similar to the original 1.0 code:

https://github.com/fastlane/fastlane-plugin-firebase_app_distribution/blob/bb0dfea89fb4cb62ce8e63722ae98afc4234280c/lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb#L21-L34

which also doesn't handle missing app_ids for android. The "if params[:app]" checks if the app id was passed in as a parameter, and we'll use it if so. Otherwise, we can try to find out the app id. I think in the future we'll have a get_app_id for android.

Reply:
Sounds good!